crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -46,29 +44,17 @@ def prepare
     end
   end
 
-  def services
-    if request.post?
-      head :not_implemented
-    else
-      render json: [], status: :not_implemented
-    end
-  end
-
   def prechecks
-    render json: @upgrade.check
+    render json: Api::Upgrade.check
   end
 
   def cancel
-    if @upgrade.cancel
+    cancel_upgrade = Api::Upgrade.cancel
+
+    if cancel_upgrade[:status] == :ok
       head :ok
     else
-      render json: { error: @upgrade.errors.full_messages.first }, status: :unprocessable_entity
+      render json: { error: cancel_upgrade[:message] }, status: cancel_upgrade[:status]
```

Comment:
Is the change from `:unprocessable_entity` really related to the switch to class methods?